### PR TITLE
Unpin `dask` & `distributed` in CI

### PR DESCRIPTION
**Pull request by galipremsagar**
Changes to be in-line with: https://github.com/rapidsai/cudf/pull/9307/

### CODE REVIEW

**Review by jakirkham — APPROVED** [review]

File: conda/recipes/dask-cuda/meta.yaml [hunk 1]
```diff
@@ -27,8 +27,8 @@ requirements:
     - setuptools
   run:
     - python
-    - dask=2021.09.1
-    - distributed=2021.09.1
+    - dask>=2021.09.1,<2021.10.0
+    - distributed>=2021.09.1,<2021.10.0
```
Comment (jakirkham) [inline conda/recipes/dask-cuda/meta.yaml:31]: Actually should we drop the upper bound?

Comment (galipremsagar) [reply conda/recipes/dask-cuda/meta.yaml:31]: ..and then only enforce upper bound:
1. During the release
2. or some necessary bug fixes or to tackle prolonged failures in CI?

Comment (jakirkham) [reply conda/recipes/dask-cuda/meta.yaml:31]: Was thinking 1. @quasiben @pentschev, do you have thoughts here?

Comment (pentschev) [reply conda/recipes/dask-cuda/meta.yaml:31]: Definitely 1. This is what we follow during development to make sure we don't lag behind Dask changes (and potential bugs).

Comment (galipremsagar) [reply conda/recipes/dask-cuda/meta.yaml:31]:
```suggestion
    - dask>=2021.09.1
    - distributed>=2021.09.1
```

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-dask==2021.09.1
-distributed==2021.09.1
+dask>=2021.09.1,<2021.10.0
+distributed>=2021.09.1,<2021.10.0
```
Comment (galipremsagar) [inline requirements.txt:2]:
```suggestion
dask>=2021.09.1
distributed>=2021.09.1
```

File: conda/recipes/dask-cuda/meta.yaml (hunk 1 above)
Comment (galipremsagar) [reply conda/recipes/dask-cuda/meta.yaml]: Done, made the changes 👍

**Review by ajschmidt8 — APPROVED** [review]

**Review by pentschev — APPROVED** [review]
Thanks @galipremsagar !

**Comment by pentschev** [comment]
@gpucibot merge

**Pull request merged**